test/passing/dune.inc

```diff
@@ -1374,6 +1374,18 @@
  (name runtest)
  (action (diff js_source.ml.ref js_source.ml.output)))
 
+(rule
+ (targets js_source.ml.ocp.output)
+ (deps .ocamlformat )
+ (action
+   (with-outputs-to %{targets}
+     (system "%{bin:ocamlformat} %{read-lines:js_source.ml.opts} %{dep:js_source.ml} | %{bin:ocp-indent} | %{bin:ocamlformat} %{read-lines:js_source.ml.opts} -"))))
```

Comment:
Why not taking `js_source.ml.output` or `js_source.ml.ref` as input for ocp-indent ?
Also, what is tested by piping ocamlformat after ocp-indent ? Stability ?
I think we are interested in the output of ocp-indent instead, if ocp-indent-compat was perfect it shouldn't change anything of the output of ocamlformat

Reply:
I tried to improve that:
- .ref file ->[ocp-indent]-> .ocp.output file
- .ocp.output file ->[promote]-> .ocp file

but dune raises an error: `exception (Invalid_argument js_source.ml.ocp.output)`

Reply:
This is not raised by Dune but by `gen.ml` in `register_file` on line 49. You need to handle the `["ocp"; "output"]` case.

Reply:
indeed, fixed thenks, maybe it's a good time to improve this error.